File: app/src/main/java/com/prox/voicechanger/media/Player.java
package com.prox.voicechanger.media;

import static com.prox.voicechanger.VoiceChangerApp.TAG;

import android.media.MediaPlayer;
import android.util.Log;

import com.prox.voicechanger.interfaces.PlayerListener;

import java.io.IOException;

public class Player implements PlayerListener {
    private MediaPlayer player;

    public Player() {
        player = new MediaPlayer();
    }

    public boolean isPlaying(){
        return player.isPlaying();
    }

    public int getCurrentPosition(){
        return player.getCurrentPosition();
    }

    public int getDuration(){
        return player.getDuration();
    }

    public void seekTo(long i){
        player.seekTo((int) i);
    }

    @Override
    public void start(String path) {
        Log.d(TAG, "Player: start " + path);
        if (player == null){
            Log.d(TAG, "Player: start null");
            return;
        }
        if (!player.isPlaying()){
            try {
                player.reset();
                player.setDataSource(path);
                player.setLooping(true);
                player.prepare();
                player.start();
            } catch (IOException e) {
                Log.d(TAG, "Player: start error " + e.getMessage());
            }
        }
    }

    @Override
    public void pause() {
        if (player == null){
            Log.d(TAG, "Player: pause null");
            return;
        }

        if (player.isPlaying()){
            player.pause();
        }
    }

    @Override
    public void resume() {
        if (player == null){
            Log.d(TAG, "Player: resume null");
            return;
        }

        if (!player.isPlaying()){
            player.start();
        }
    }

    @Override
    public void stop() {
        if (player == null){
            Log.d(TAG, "Player: stop null");
            return;
        }

        if (player.isPlaying()){
            player.stop();
        }
    }

    @Override
    public void release() {
        if (player == null){
            Log.d(TAG, "Player: release null");
            return;
        }
        if (player.isPlaying()){
            player.stop();
        }
        player.release();
        player = null;
        Log.d(TAG, "Recorder: release");
    }
}

File: app/src/main/java/com/prox/voicechanger/media/Player.kt
package com.prox.voicechanger.media

import com.prox.voicechanger.interfaces.PlayerListener
import android.media.MediaPlayer
import com.prox.voicechanger.VoiceChangerApp
import android.media.AudioManager
import android.util.Log
import java.io.IOException

class Player : PlayerListener {
    private var player: MediaPlayer?
    val isPlaying: Boolean
        get() = player!!.isPlaying
    val currentPosition: Int
        get() = player!!.currentPosition
    val duration: Int
        get() = player!!.duration

    fun seekTo(i: Long) {
        player!!.seekTo(i.toInt())
    }

    override fun setNewPath(path: String?) {
        Log.d(VoiceChangerApp.TAG, "Player: setNewPath $path")
        if (player == null) {
            Log.d(VoiceChangerApp.TAG, "Player: player null")
            return
        }
        try {
            player!!.reset()
            player!!.setDataSource(path)
            player!!.setAudioStreamType(AudioManager.STREAM_MUSIC)
            player!!.isLooping = true
            player!!.prepare()
        } catch (e: IOException) {
            Log.d(VoiceChangerApp.TAG, "Player: setNewPath error " + e.message)
        }
    }

    override fun start() {
        if (player == null) {
            Log.d(VoiceChangerApp.TAG, "Player: player null")
            return
        }
        player!!.start()
    }

    override fun pause() {
        if (player == null) {
            Log.d(VoiceChangerApp.TAG, "Player: player null")
            return
        }
        if (player!!.isPlaying) {
            player!!.pause()
        }
    }

    override fun resume() {
        if (player == null) {
            Log.d(VoiceChangerApp.TAG, "Player: player null")
            return
        }
        if (!player!!.isPlaying) {
            player!!.start()
        }
    }

    override fun stop() {
        if (player == null) {
            Log.d(VoiceChangerApp.TAG, "Player: player null")
            return
        }
        if (player!!.isPlaying) {
            player!!.stop()
        }
    }

    override fun release() {
        if (player == null) {
            Log.d(VoiceChangerApp.TAG, "Player: player null")
            return
        }
        if (player!!.isPlaying) {
            player!!.stop()
        }
        player!!.release()
        player = null
        Log.d(VoiceChangerApp.TAG, "Player: release")
    }

    init {
        player = MediaPlayer()
    }
}